src/algo.cpp

```diff
@@ -0,0 +1,230 @@
+#include "algo.h"
+#include "game_battler.h"
+#include "game_actor.h"
+#include "game_enemy.h"
+#include "attribute.h"
+#include "player.h"
+#include <lcf/rpg/skill.h>
+
+#include <algorithm>
+
+namespace Algo {
+
+bool IsRowAdjusted(const Game_Actor& actor, lcf::rpg::System::BattleCondition cond, bool offense) {
+	return (cond == lcf::rpg::System::BattleCondition_surround
+			|| (actor.GetBattleRow() == (1 - offense)
```

Comment:
> Use Algo library and port normal attack calculations to it 

``(1 - offense)``. That's a bool, how about ``actor.GetBattleRow() != offense``? Or whatever the intention is (have unit tests ;)). Is hard to read.